src/file_utils.cr

```diff
@@ -305,16 +305,21 @@ module FileUtils
   end
 
   # Moves *src_path* to *dest_path*.
+  # If src_path and dest_path exist on the different disk partition,
+  # the file is copied then the original file is removed.
   #
   # ```
   # require "file_utils"
   #
   # FileUtils.mv("afile", "afile.cr")
   # ```
   #
-  # NOTE: Alias of `File.rename`
   def mv(src_path : String, dest_path : String) : Nil
     File.rename(src_path, dest_path)
+  rescue ex : File::Error
+    raise ex unless Errno.value.in?(Errno::EXDEV, Errno::EPERM)
```

Comment:
Yeah totally agree, but thought that change might cause regression, as don't know if any piece of code is relying on exception being raised by `rename` method, 2nd thought was that changing the behavior will make it different from other methods which are raising the exceptions. 

I can try to dig deeper into the code-base to see if there are any areas which are relying on such behavior, if none, I believe it make more sense to avoid unnecessary stack trace.

Reply:
`Crystal::System::File.rename` is an internal method. It can be changed easily to return the error instead of raising it directly.

`File.rename` is the public API and would retain its behaviour like this:
```cr
def self.rename(old_filename : Path | String, new_filename : Path | String) : Nil
  if error = Crystal::System::File.rename(old_filename.to_s, new_filename.to_s)
    raise error
  end
end
```

There should be no other direct calls to `Crystal::System::File.rename` in the codebase.

Reply:
Pushed changes to this PR.